### PR TITLE
Add Ender Dragon Respawn & Tetra Forge Hammer Ponders

### PR DESCRIPTION
**Pull request by TemperedFool**
This adds the two following ponders:

[Ender Dragon Respawn](https://cdn.discordapp.com/attachments/884949955410796564/887026299250561044/dragon_respawn.mov)
[Tetra Forge Hammer](https://cdn.discordapp.com/attachments/884949955410796564/887360564312023141/forge_hammer.mov)

I've also taken the opportunity (before things get out of hand) to refactor Ponder's base.js - I've moved the item registration for the categories from each of the ponder scripts to base.js, and moved all of the category registration with it to a constant structure. Should make maintenance easier.

### CODE REVIEW

**Comment by zedekblue** [comment]
Bless you for making one for the forge hammer -- I spent a good hour figuring that one out with outdated resources linked on the mod page.

File: kubejs/client_scripts/ponder/base.js [hunk 1]
```diff
@@ -1,22 +1,50 @@
...
-      'industrialforegoing:latex_processing_unit',
-      'Industrial Foregoing',
-      'Learn more about the Industrial Foregoing mod and its intriguing mechanics!'
-    );
-    event.create(
-      'enigmatica:basics',
+const categories = [
+  {
+    id: 'enigmatica:industrial_foregoing',
+    cat_item_id:'industrialforegoing:latex_processing_unit',
```
Comment (NillerMedDild) [inline kubejs/client_scripts/ponder/base.js:6]: please be explicit with variable names, `category_item_id` would be better imo :)

File: kubejs/client_scripts/ponder/base.js [hunk 2]
```diff
@@ -1,22 +1,50 @@
...
+    title:'Industrial Foregoing',
+    desc:'Learn more about the Industrial Foregoing mod and its intriguing mechanics!',
```
Comment (NillerMedDild) [inline kubejs/client_scripts/ponder/base.js:8]: same for this one, no need to keep it short

File: kubejs/client_scripts/ponder/enigmatica/forge_hammer.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          .showText(60)
+          .colored(PonderPalette.WHITE)
+          .text(`Using the Forge Hammer will drain charge from the Magmatic Cells, which will need to be replaced when depleted.`)
+          .independent(0);
+          scene.idle(35);
+          scene
+          .overlay()
+          .showText(60)
+          .colored(PonderPalette.WHITE)
+          .text(`The Forge Hammer can be found in word in underground Tetra, frequently as part of a functional multiblock.`)
```
Comment (NillerMedDild) [inline kubejs/client_scripts/ponder/enigmatica/forge_hammer.js:82]: I suggest we change this to something like `The Forge Hammer can be found in Tetra structures, found underground in cold biomes.`

File: kubejs/client_scripts/ponder/base.js [hunk 4]
```diff
@@ -1,22 +1,50 @@
...
-onEvent('ponder.tag.registry', (event) => {
-    event.create(
-      'enigmatica:industrial_foregoing',
-      'industrialforegoing:latex_processing_unit',
-      'Industrial Foregoing',
-      'Learn more about the Industrial Foregoing mod and its intriguing mechanics!'
-    );
-    event.create(
-      'enigmatica:basics',
+const categories = [
```
Comment (NillerMedDild) [inline kubejs/client_scripts/ponder/base.js:3]: I don't see the new entries here?

**Review by TemperedFool — COMMENTED** [review]
Ok, the two specific changes should be addressed now.

**Comment by NillerMedDild** [comment]
Nice! Only one to go :D

File: kubejs/client_scripts/ponder/enigmatica/forge_hammer.js (hunk 3 above)
Comment (TemperedFool) [reply kubejs/client_scripts/ponder/enigmatica/forge_hammer.js]: Done!

**Review by TemperedFool — COMMENTED** [review]
I'll get used to the review interface some day! Missed that one before, resolved now.

**Pull request merged**